features/step_definitions/collection.js

```diff
@@ -0,0 +1,267 @@
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.firstCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    Given(/^they have a collection "anotherCollection"$/, function step() {
+        return createCollection.call(this, { name: 'anotherCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.secondCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    When(/^they fetch all their collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+        });
+    });
+
+    Then(/^they can see those collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.body.length, 2);
+            Assert.strictEqual(response.body[0].name, 'anotherCollection');
+            Assert.strictEqual(response.body[1].name, 'myCollection');
+        });
+    });
+
+    When(/^they delete that collection$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'DELETE',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.oneOf(response.statusCode, [404, 204]);
+        });
+    });
+
+    Then(/^that collection no longer exists$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 404);
+        });
+    });
+
+    Then(/^that pipeline still exists$/, function step() {
+        return this.ensurePipelineExists({ repoName: this.repoName });
```

Comment:
This should not use the `this.ensurePipelineExists`, which creates a pipeline if missing. This should be making a call for the specific pipeline and not receive a `404`.